src/hooks/usePickerInput.ts

```diff
@@ -43,43 +49,53 @@ export default function usePickerInput({
       triggerOpen(true);
     },
     onKeyDown: e => {
-      switch (e.which) {
-        case KeyCode.ENTER: {
-          if (!open) {
-            triggerOpen(true);
-          } else if (onSubmit() !== false) {
-            setTyping(true);
-          }
+      if (onKeyDown) {
+        const preventDefaultBehaviors = (): void => {
+          setPreventDefault(true);
+        };
 
-          e.preventDefault();
-          return;
-        }
+        onKeyDown(e, preventDefaultBehaviors);
+      }
 
-        case KeyCode.TAB: {
-          if (typing && open && !e.shiftKey) {
-            setTyping(false);
-            e.preventDefault();
-          } else if (!typing && open) {
-            if (!forwardKeyDown(e) && e.shiftKey) {
+      if (preventDefault === false) {
```

Comment:
```suggestion
      if (preventDefault === true) return
```

This can make the changes smaller.

Reply:
ok

Reply:
``` js
if (preventDefault === true) return
```

This is better 😄

Reply:
> > @conquera99 can you resolve conflicts?
> 
> okay, i will try to resolve the conflicts tonight

@kerm1it hi, there is some error when i try to commit changes (changing preventdefault state to ref and add test case), is this error just in my pc?

```error  Parse errors in imported module '../src/interface': Cannot read property 'map' of undefined```
```error  Parsing error: Cannot read property 'map' of undefined```

Reply:
You can run `npm run lint`

Reply:
thank you!